test/e2e_test.go

```diff
@@ -2524,3 +2541,303 @@ func TestOffline(t *testing.T) {
+	_, _, cleanup1 := mkimage(t, signedImg1)
+	defer cleanup1()
+	signedImg2 := path.Join(repo, "cosign-e2e-dockerfile-signed2")
+	_, _, cleanup2 := mkimage(t, signedImg2)
+	defer cleanup2()
+	unsignedImg := path.Join(repo, "cosign-e2e-dockerfile-unsigned")
+	_, _, cleanupUnsigned := mkimage(t, unsignedImg)
+	defer cleanupUnsigned()
+
+	// sign the images using --identity-token
+	ko := options.KeyOpts{
+		FulcioURL:        fulcioURL,
+		RekorURL:         rekorURL,
+		IDToken:          identityToken,
+		SkipConfirmation: true,
+	}
+	so := options.SignOptions{
+		Upload:           true,
+		SkipConfirmation: true,
+	}
+	ctx := context.Background()
+	must(sign.SignCmd(ro, ko, so, []string{signedImg1}), t)
+	must(sign.SignCmd(ro, ko, so, []string{signedImg2}), t)
+
+	// create the dockerfiles
+	singleStageDockerfileContents := fmt.Sprintf(`
+FROM %s
+`, signedImg1)
+	singleStageDockerfile := mkfile(singleStageDockerfileContents, td, t)
+
+	unsignedBuildStageDockerfileContents := fmt.Sprintf(`
+FROM %s
+
+FROM %s
+
+FROM %s
+`, signedImg1, unsignedImg, signedImg2)
+	unsignedBuildStageDockerfile := mkfile(unsignedBuildStageDockerfileContents, td, t)
+
+	fromAsDockerfileContents := fmt.Sprintf(`
+FROM --platform=linux/amd64 %s AS base
+`, signedImg1)
+	fromAsDockerfile := mkfile(fromAsDockerfileContents, td, t)
+
+	withArgDockerfileContents := `
+ARG test_image
+
+FROM ${test_image}
+`
+	withArgDockerfile := mkfile(withArgDockerfileContents, td, t)
+
+	withLowercaseDockerfileContents := fmt.Sprintf(`
+from %s
+`, signedImg1)
+	withLowercaseDockerfile := mkfile(withLowercaseDockerfileContents, td, t)
+
+	issuer := os.Getenv("OIDC_URL")
+
+	tests := []struct {
+		name       string
+		dockerfile string
+		baseOnly   bool
+		env        map[string]string
+		wantErr    bool
+	}{
+		{
+			name:       "verify single stage",
+			dockerfile: singleStageDockerfile,
+		},
+		{
+			name:       "verify unsigned build stage",
+			dockerfile: unsignedBuildStageDockerfile,
+			wantErr:    true,
+		},
+		{
+			name:       "verify base image only",
+			dockerfile: unsignedBuildStageDockerfile,
+			baseOnly:   true,
+		},
+		{
+			name:       "verify from as",
+			dockerfile: fromAsDockerfile,
+		},
+		{
+			name:       "verify with arg",
+			dockerfile: withArgDockerfile,
+			env:        map[string]string{"test_image": signedImg1},
+		},
+		{
+			name:       "verify image exists but is unsigned",
+			dockerfile: withArgDockerfile,
+			env:        map[string]string{"test_image": unsignedImg},
+			wantErr:    true,
+		},
+		{
+			name:       "verify with lowercase",
+			dockerfile: withLowercaseDockerfile,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cmd := dockerfile.VerifyDockerfileCommand{
+				VerifyCommand: cliverify.VerifyCommand{
+					CertVerifyOptions: options.CertVerifyOptions{
+						CertOidcIssuer: issuer,
+						CertIdentity:   certID,
+					},
+					RekorURL:   rekorURL,
+					IgnoreTlog: true,
```

Comment:
Since the tests are now hermetic, can we remove `IgnoreTlog` so it tests inclusion in the log (and for the other new tests)? I would expect the test to still pass.

Reply:
Removed `IgnoreTlog`, but also had to add `TlogUpload` to the sign command.